File: Money.API/Models/SumByCategory.cs
namespace Money.API.Models
{
    public class SumByCategory
    {
        public string Name { get; set; }

        public decimal Amount { get; set; }
        public decimal Percentage { get; set; }

        public IEnumerable<SumByCategory> SubCategory { get; set; }
    }
}

File: Money/src/app/models/sum-by-category.ts
/**
 * Sum of credit et debit for the month by category.
 */
export interface SumByCategory {
    /**
     * Name of the category
     */
    name: string;

    /**
     * Transactions sum for the category.
     * Negative if it's debit, positive it's credit.
     */
    amount: number;

    /**
     * Percentage of the category for the month
     */
    percentage: number;

    /**
     * Sum of credit et debit for the month by sub category.
     */
    subCategory: SumByCategory[];
}